File: src/client_gen/javascript/my-strategy.js
const model = require('./model/index');

class MyStrategy {
    async getAction(playerView, debugInterface) {
        throw "Write your strategy here";
    }
    async debugUpdate(playerView, debugInterface) {
        await debugInterface.send(new model.DebugCommand.Clear());
        await debugInterface.getState();
    }
}

module.exports.MyStrategy = MyStrategy;

File: src/client_gen/rust/my_strategy.rs
use super::Debug;

pub struct MyStrategy {}

impl MyStrategy {
    pub fn new() -> Self {
        Self {}
    }
    pub fn get_action(
        &mut self,
        player_view: &model::PlayerView,
        debug_interface: &mut DebugInterface,
    ) -> model::Action {
        todo!()
    }
    pub fn debug_update(
        &mut self,
        player_view: &model::PlayerView,
        debug_interface: &mut DebugInterface,
    ) {
    }
}
